### PR TITLE
Add 'date' param to Ranking query

### LINKED ISSUE #11: Criar endpoint de Ranking por Data
Adicionar data no ranking, e no endpoint retornar ranking por trilha/data

### PR DESCRIPTION
**Pull request by bhabermann**
This commit solves the problem of fetching all TrackRankings when
the user wants to see only the ranking of one day.

Resolves #11

### CODE REVIEW

File: src/DexQuiz.Core/Services/RankingService.cs [hunk 1]
```diff
@@ -112,11 +111,18 @@ public async Task UpdateRankingAfterUserAnswerAsync(AnsweredQuestion answeredQue
...
-        private async Task<IEnumerable<TrackRanking>> GetOrderedTrackRankingsAsync(int trackId) =>
-            (await _trackRankingRepository.FindAsync(r => r.TrackId == trackId && r.CompletedTime != null))
-                                              .OrderByDescending(r => r.Points)
-                                              .ThenBy(r => r.CompletedTime)
-                                              .Select(MapTrackRankingInsertingPosition);
+        private async Task<IEnumerable<TrackRanking>> GetOrderedTrackRankingsAsync(int trackId, DateTime? date)
+        {
+            IEnumerable<TrackRanking> trackRankings = null;
+            if (date.HasValue)
+                trackRankings = await _trackRankingRepository.FindAsync(r => r.TrackId == trackId && r.CompletedTime != null && r.StartedAtUtc.Date == date);
```
Comment (gustavobigardi) [inline src/DexQuiz.Core/Services/RankingService.cs:118]: Aqui não sera melhor comparar com date.Date também? pra garantir....

**Review by gustavobigardi — COMMENTED** [review]
Adicionei um comentario para verificar.

**Review by Gabriel-Cordeiro — CHANGES_REQUESTED** [review]
Hey @bhabermann as we talked last week, all tasks should have their related unit tests, can you do it and update this PR? tnks.

File: src/DexQuiz.Core/Services/RankingService.cs (hunk 1 above)
Comment (bhabermann) [reply src/DexQuiz.Core/Services/RankingService.cs:118]: Não havia pensado nessa hipótese, mas acredito que seja melhor sim. Vou fazer a alteração;

**Comment by bhabermann** [comment]
> Hey @bhabermann as we talked last week, all tasks should have their related unit tests, can you do it and update this PR? tnks.

I'll create the test in a next PR to solve the issue #6

**Review by Gabriel-Cordeiro — APPROVED** [review]

**Pull request merged**